test/parallel/test-http2-client-rststream-before-connect.js

```diff
@@ -56,12 +61,6 @@ server.listen(0, common.mustCall(() => {
     client.close();
   }));
 
-  req.on('error', common.expectsError({
-    code: 'ERR_HTTP2_STREAM_ERROR',
-    name: 'Error',
-    message: 'Stream closed with error code NGHTTP2_PROTOCOL_ERROR'
-  }));
-
```

Comment:
I don't think this should be removed

Reply:
@mcollina This is a result of the fix for making `.destroy()` not throw when sending RST. It throws only when receiving RST. This means that `stream.close(1)` won't emit error. Are you okay with this change? By looking at the original code for `close()` it looks like the author didn't mean to emit error on `close`.

Otherwise I would have to make `stream.close()` set `stream.rstCode` and then call `this.destroy(error)`.

Reply:
Also, should `session.destroy()` make its streams error or should they exit cleanly?

Reply:
Actually my understanding was wrong. It shouldn't be removed indeed.

It should error via `onStreamClose` (so pipeline fails etc.) but fails to do so because https://github.com/nodejs/node/blob/56dea478d293a0a3f562dbefacddea9a4583f73d/lib/internal/http2/core.js#L2400

gets called first.

Reply:
I'm not fluent in stream internals but `setImmediate` seems to work here (in afterShutdown), so I went with that. 

Looks like the main cause is that writable shutdown is delayed by single tick in order to attach the END_STREAM flag. It's a pity nghttp2 has no API to amend the last queued DATA frame.